src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -107,6 +110,94 @@ protected int read(ByteBuffer bb)
         return ChannelInputStream.read(ch, bb, true);
     }
 
+    @Override
+    public byte[] readAllBytes() throws IOException {
+        if (!(ch instanceof SeekableByteChannel))
+            return super.readAllBytes();
+        SeekableByteChannel sbc = (SeekableByteChannel)ch;
+
+        long length = sbc.size();
+        long position = sbc.position();
+        long size = length - position;
+
+        if (length <= 0 || size <= 0)
+            return super.readAllBytes();
+
+        if (size > (long) Integer.MAX_VALUE) {
```

Comment:
```suggestion
        else if (size > (long) Integer.MAX_VALUE) {
```

Reply:
It is common practice in the libraries to use multiple `if` statements to return as soon as feasible.